### DISCUSSION TITLE
Is there any method to map keys to functions now?

### DISCUSSION THREAD

**Discussion by Tgist1024** [ayamir/nvimdots#515] [Q&A] [answered]
I remember using keymap to call function was possible in a former version.
Now I want to add a plugin that has a quick setup like this:
```lua
local ht = require('haskell-tools')
local def_opts = { noremap = true, silent = true, }
ht.setup {
  hls = {
    on_attach = function(client, bufnr)
      local opts = vim.tbl_extend('keep', def_opts, { buffer = bufnr, })
      -- haskell-language-server relies heavily on codeLenses,
      -- so auto-refresh (see advanced configuration) is enabled by default
      vim.keymap.set('n', '<space>ca', vim.lsp.codelens.run, opts)
      vim.keymap.set('n', '<space>hs', ht.hoogle.hoogle_signature, opts)
    end,
  },
}
-- Suggested keymaps that do not depend on haskell-language-server
-- Toggle a GHCi repl for the current package
vim.keymap.set('n', '<leader>rr', ht.repl.toggle, def_opts)
-- Toggle a GHCi repl for the current buffer
vim.keymap.set('n', '<leader>rf', function()
  ht.repl.toggle(vim.api.nvim_buf_get_name(0))
end, def_opts)
vim.keymap.set('n', '<leader>rq', ht.repl.quit, def_opts)
```
It seems present keymap functions like `map_cu`, `map_cr` or any others could not call functions like above. All of them do is just pass a string into the same function.
So I need some help to solve this.
Or maybe what I should do is just put the setup in my custom config? (Seems not bad for the ease)

**Comment by charliie-dev** [comment, marked as answer]
https://github.com/ayamir/nvimdots/blob/b3fb4651709ffd60a8edd3adc5543493e7087094/lua/keymap/bind.lua#L137-L142 is how u set functions to keymaps.

Here's an example:

https://github.com/ayamir/nvimdots/blob/b3fb4651709ffd60a8edd3adc5543493e7087094/lua/keymap/tool.lua#L129-L134

As u can see it's contains a function.

**Reply by Tgist1024** [reply]
Thanks a lot!
